packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -0,0 +1,88 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:io';
+
+import '../doctor_validator.dart';
+
+class WindowsVersionValidator extends DoctorValidator {
+  const WindowsVersionValidator() : super('Windows Version');
+  @override
+  Future<ValidationResult> validate() async {
+    final ProcessResult result = await Process.run('systeminfo', <String>[]);
+    final String resultStdout = result.stdout as String;
```

Comment:
You need to check if `ProcessResult.exitCode` was successful before going further

Reply:
Oh good point, I am adding a try, catch to find any general ProcessExceptions now and then also doing a check afterwords on the exit code itself